src/backend/renderer/mod.rs

```diff
@@ -757,3 +757,26 @@ pub fn buffer_dimensions(buffer: &wl_buffer::WlBuffer) -> Option<Size<i32, Buffe
         Err(_) => None,
     }
 }
+
+/// Returns if the underlying buffer is y-inverted
+///
+/// *Note*: This will only return y-inverted for buffer types known to smithay (see [`buffer_type`])
+#[cfg(feature = "wayland_frontend")]
+pub fn buffer_y_inverted(buffer: &wl_buffer::WlBuffer) -> Option<bool> {
```

Comment:
The way this is used, it should be sufficient to just return bool instead of Option<bool>. The None at the bottom is replaced by false and you can drop the Some returns.

Reply:
If the buffer is not supported, probably just use a Result<book, UnmanagedResource>

Reply:
> The way this is used, it should be sufficient to just return bool instead of Option. The None at the bottom is replaced by false and you can drop the Some returns.

The Option is used to communicate that the buffer is not managed by smithay and we can't tell if it is inverted. Anvil only uses smithay provided buffers, but downstream could introduce other buffer types.

Reply:
> If the buffer is not supported, probably just use a `Result<bool, UnmanagedResource>`

Yes, I choose Option to align it with the other already existing buffer functions, all of them return an Option in that case. But I am not opposed to change it to a Result, we just should stay consistent.

Reply:
For what it is worth, the `Result` is more self-explanatory. (and thus probably the better fit here)

Otherwise the function needs some docs, what `None` means.

Reply:
That comment is sufficient. I would prefer all function to be changed, given we have a proper error type - that was largely my oversight when writing these functions, I think - but that doesn't necessarily need to be part of the PR. I agree consistency is key here.